client/app/main/student/student.controller.js

```diff
@@ -109,6 +109,50 @@ function StudentCtrl($scope, $stateParams, Intervention, Modal, Outreach,
       toastr.error(err);
     });
   };
+  $scope.toggleArchiveOutreach = function(outreach) {
+    var oldValue = outreach.archived;
+    var promise = Outreach.toggleArchive({
+      id: outreach._id
+    }, {
+      archived: !(outreach.archived)
+    }).$promise;
+    promise.then(function(outreach) {
+      $scope.$evalAsync(function() {
+        var outreaches = $scope.student.outreaches
+        outreaches.map(function(oldOutreach, i) {
```

Comment:
Avoid using `.map` like this. In this case you don't want to return an array that is a map of values to the original array, you want to iterate over each element of the array using [`array.forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach). 

```
outreaches.forEach(function(oldOutreach, i) {
  // Do stuff
});
```